### PR TITLE
Nyuntam vision Integration

### PR DESCRIPTION
**Pull request by Akash-guna**

### CODE REVIEW

File: factory.py [hunk 1]
```diff
@@ -101,7 +105,7 @@ def create_from_json(
 
         with open(path, "r") as f:
             args = json.load(f)
-
+        set_logger(logging_path=args['LOGGING_PATH'])
```
Comment (shwu-nyunai) [inline factory.py:108]: why do we need to set the logger here?

Comment (Akash-guna) [reply factory.py:108]: I have path issues when importing it in my factory so I included it in the main factory itself.

Comment (Akash-guna) [reply factory.py]: yeah made similar change

File: utils/vision_utils.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+import os
+from typing import List
+
+logger = logging.getLogger(__name__)
+
+
+def validate(val_loader, model, criterion, args):
+    from trailmet.utils import accuracy, AverageMeter
+    import torch
+    logger = logging.getLogger(__name__)
```
Comment (shwu-nyunai) [inline utils/vision_utils.py:13]: we need not create another logger instance. we already have one instance at line 7. one of these can be removed.

File: utils/vision_utils.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        ]
+    )
+    os.makedirs(f"{os.getcwd()}/logs/Validate", exist_ok=True)
+    logger_file = f"{os.getcwd()}/logs/Validate/{name}.log"
+    logging.basicConfig(
+        filename=logger_file,
+        format="%(asctime)s - %(levelname)s - %(name)s - %(message)s",
+        datefmt="%m/%d/%Y %H-%M-%S",
+        level=logging.INFO,
+    )
```
Comment (shwu-nyunai) [inline utils/vision_utils.py:27]: we can use set logger if needed but I assume this is an intermediate step (within vision) and a logger is expected to have been initialised.

**Review by shwu-nyunai — CHANGES_REQUESTED** [review]
can we also run a `black . --exclude <vision> --exclude <text_generation>` in the root and format everything once.

File: utils/vision_utils.py (hunk 3 above)
Comment (Akash-guna) [reply utils/vision_utils.py]: wasnt using that file removed completely

**Review by shwu-nyunai — APPROVED** [review]
lgtm. thanks @Akash-guna 
1 last question - we ran this via nyuntam as well right?

**Pull request merged**